backend/tilavarauspalvelu/integrations/keyless_entry/client.py

```diff
@@ -306,7 +306,7 @@ def change_reservation_access_code(cls, reservation: Reservation | uuid.UUID) ->
 
         data = cls.response_json(response)
         parsed_data = cls._parse_access_code_modify_response(data)
-        cls._update_cached_reservation_response(parsed_data, ext_uuid=reservation_uuid)
+        cls._clear_cached_reservation_response(ext_uuid=reservation_uuid)
```

Comment:
Clear cache should be called between `_validate_reservation_response` and `response_json`, since at that point we know there is a 200 OK response, and the access code was changed, but we might still fail parsing the response.